src/Microsoft.Health.Fhir.Liquid.Converter.UnitTests/Filters/SegmentFiltersTests.cs

```diff
@@ -116,6 +117,72 @@ public void GivenAnHl7v2Data_WhenHasSegments_CorrectResultShouldBeReturned()
+            Assert.Equal(expectedPidValue, pidSlicedData[0].Value);
+            Assert.Equal(4, pidSlicedData[0].Meta.Count);
+            Assert.Equal(4, pidSlicedData[0].Data.Count);
+            Assert.Equal("PID", pidSlicedData[0].Meta[0]);
+            Assert.Equal("PD1", pidSlicedData[0].Meta[1]);
+            Assert.Equal(@"PD1|||||||||||02^Reminder/recall - any method^HL70215|N|20150624|||A|19830615|20150624", pidSlicedData[0].Data[1].Value);
+
+            // Sliced data could only contain single segment and endSegment cound be null
+            List<Hl7v2Data> obxSlicedData = Filters.SliceDataBySegments(TestData, segmentList["OBX"]);
+            Assert.Equal(4, obxSlicedData.Count);
+            Assert.Single(obxSlicedData[0].Meta);
+            Assert.Single(obxSlicedData[1].Meta);
+            Assert.Single(obxSlicedData[2].Meta);
+            Assert.Equal(5, obxSlicedData[3].Meta.Count);
+            Assert.Single(obxSlicedData[0].Data);
+            Assert.Single(obxSlicedData[1].Data);
+            Assert.Single(obxSlicedData[2].Data);
+            Assert.Equal(5, obxSlicedData[3].Data.Count);
+            Assert.Equal("OBX", obxSlicedData[0].Meta[0]);
+            Assert.Equal(@"OBX|1|CE|30963-3^Vaccine Funding Source^LN|1|PHC70^Private^CDCPHINVS||||||F|||20150624", obxSlicedData[0].Data[0].Value);
+            string expectedObxValue = @"OBX|1|CE|30963-3^Vaccine Funding Source^LN|1|PHC70^Private^CDCPHINVS||||||F|||20150624" + System.Environment.NewLine;
+            Assert.Equal(expectedObxValue, obxSlicedData[0].Value);
+
+            // Sliced data could contain multiple segments
+            List<Hl7v2Data> orcSlicedData = Filters.SliceDataBySegments(TestData, segmentList["ORC"]);
+            Assert.Equal(3, orcSlicedData.Count);
+            Assert.Equal(7, orcSlicedData[0].Meta.Count);
+            Assert.Equal(2, orcSlicedData[1].Meta.Count);
+            Assert.Equal(2, orcSlicedData[2].Meta.Count);
+            Assert.Equal(7, orcSlicedData[0].Data.Count);
+            Assert.Equal(2, orcSlicedData[1].Data.Count);
+            Assert.Equal(2, orcSlicedData[2].Data.Count);
+            Assert.Equal("ORC", orcSlicedData[0].Meta[0]);
+            string expectedOrcValue = @"ORC|RE|4422^NIST-AA-IZ-2|13696^NIST-AA-IZ-2|||||||7824^Jackson^Lily^Suzanne^^^^^NIST-PI-1^L^^^PRN||654^Thomas^Wilma^Elizabeth^^^^^NIST-PI-1^L^^^MD|||||NISTEHRFAC^NISTEHRFacility^HL70362|";
+            Assert.Equal(expectedOrcValue, orcSlicedData[0].Data[0].Value);
+
+            // Endsegment precedes the last object in the segmentList
```

Comment:
Nit: It would be better to add some comments about what result will be returned in this case.